src/vs/workbench/contrib/format/browser/formatActionsNone.ts

```diff
@@ -68,7 +68,7 @@ registerEditorAction(class FormatDocumentMultipleAction extends EditorAction {
 			const res = await dialogService.show(
 				Severity.Info,
 				message,
-				[nls.localize('cancel', "Cancel"), nls.localize('install.formatter', "Install Formatter...")]
+				[nls.localize('install.formatter', "Install Formatter..."), nls.localize('cancel', "Cancel")]
```

Comment:
same - `choice === 1` assumes you want to format. This is now `0`, right?

Reply:
yeah, better to set `cancelId`.

Tbh this is Electron API that we basically just have exposed 1 to 1 with the way Electron decided to deal with cancellation. From the docs:

```ts
/**
 * The index of the button to be used to cancel the dialog, via the `Esc` key. By
 * default this is assigned to the first button with "cancel" or "no" as the label.
 * If no such labeled buttons exist and this option is not set, `0` will be used as
 * the return value.
 */
cancelId?: number;
```

Esp. the part around checking for a "label" is bad when we think about the fact that our dialogs get translated to other languages. 

I can do a pass to check for other cases where we do not set `cancelId` and maybe make that even a required property.

Reply:
via https://github.com/microsoft/vscode/pull/155049/commits/517600ee1f99f27abd202e62d69ecc600db20c68

Reply:
Yeah, similar to how we use `IAction` in notifications:

https://github.com/microsoft/vscode/blob/c9c6219f67ff76f10fe548bc1c5247ed3a91577a/src/vs/platform/notification/common/notification.ts#L127-L143